### PR TITLE
Udpate `timm` & `torchmetrics` dependency

### LINKED ISSUE #639: Update torchmetrics dependency
we fail to push to pypy because of the torchmetrics depedency on git version;

they pushed a new one: https://github.com/Lightning-AI/torchmetrics/issues/2734

### PR DESCRIPTION
**Pull request by ioangatop**
Closes #639

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -45,7 +45,7 @@ dependencies = [
     "onnx>=1.16.0",
     "toolz>=0.12.1",
     "rich>=13.7.1",
-    "torchmetrics @ git+https://github.com/Lightning-AI/torchmetrics.git",
+    "torchmetrics>=1.4.2",
```
Comment (nkaenzig) [inline pyproject.toml:48]: should we maybe also update the `timm` version? Currently we point to main, but I think in the current release the features we need are available

Comment (ioangatop) [reply pyproject.toml:48]: yeap was thinking the same..

File: src/eva/vision/models/networks/backbones/pathology/kaiko.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ def kaiko_vitl14(
     Returns:
         The model instance.
     """
-    return torch.hub.load(
+    return torch.hub.load(  # type: ignore
```
Comment (nkaenzig) [inline src/eva/vision/models/networks/backbones/pathology/kaiko.py:117]: why are the `type: ignore` required here?

**Review by nkaenzig — APPROVED** [review]

**Comment by ioangatop** [comment]
waiting due to: https://github.com/Lightning-AI/torchmetrics/issues/2734

**Pull request merged**